ui/core/src/model/calculations.ts

```diff
@@ -55,56 +56,151 @@ export const CALCULATIONS_CONFIG: Readonly<Record<CalculationType, CalculationCo
     label: 'Sum',
     description: 'The sum of all values',
   },
+  Min: {
+    label: 'Min',
+    description: 'Minimum value',
+  },
+  Max: {
+    label: 'Max',
+    description: 'Maximum value',
+  },
 } as const;
 
 export const DEFAULT_CALCULATION: CalculationType = 'Sum';
 
 type CalculationValue = number | null | undefined;
 
+/**
+ * Calculate a multiple values for a set of time series data.
+ *
+ * @param values - Array of time series data.
+ * @param includeCalculations - Array of calculations to include.
+ */
+export function getCalculations<IncludeCalcs extends CalculationType[]>(
+  values: TimeSeriesValueTuple[],
+  includeCalculations: IncludeCalcs
+): Record<
+  // This extract combined with the generics above keeps the key of the returned
+  // record to *just* the specified calculations.
+  Extract<CalculationType, IncludeCalcs[number]>,
+  CalculationValue
```

Comment:
In other words, if you request `['First', 'Last']`, the return type would be `Record<'First' | 'Last', CalculationValue>`